python/lsst/pipe/tasks/configurableActions/_configurableAction.py

```diff
@@ -45,5 +48,17 @@ class ConfigurableAction(Config):
     it will raise a `NotImplementedError` when called. Subclasses that
     represent concrete actions must provide an override.
     """
+
+    identity: str | None = None
+    """If a configurable action is assigned to a `ConfigurableActionField`, or a
+    `ConfigurableActionStructField` the name of the field will be bound to this
+    variable when it is retrieved.
+    """
+
+    def __setattr__(self, attr, value, at=None, label="assignment"):
+        if attr == 'identity':
+            return object.__setattr__(self, attr, value)
+        return super().__setattr__(attr, value, at, label)
+
     def __call__(self, *args, **kwargs) -> Any:
```

Comment:
This is preexisting, and it seems like something we might have even discussed already, but doesn't this signature imply that all `ConfigurableAction` subclasses *must* accept arbitrary `*args, **kwargs`, rather than the concrete signature declared by some intermediate `ConfigurableAction` abstract subclass?  (This is exactly the same problem as `Task.run`, but without the history that makes that one harder to remove, I think).

In any case, it seems to me to make more sense to drop the `__call__` here and define it only in subclasses like `AnalysisAction` that have a concrete signature.  If the goal here is to be able to define some concrete `ConfigurableAction` subclasses here that have *generic* `__call__` signatures becaus they're just forwarders, I think there's a better way to type that with PEP-612 stuff, though I don't know more than that.

Reply:
I will look more into PEP-612 stuff before merging this but AFAICT type checkers have no problem with this from a substitution principal, and I am not sure why. I dont know if they treat `__` methods as special, or if `__call__` is special because it is used in construction in so many parts of the language.